src/devices/hubled.ts

```diff
@@ -20,13 +20,10 @@ export class HubLED extends Device {
      */
     public setColor (color: number | boolean) {
         return new Promise((resolve, reject) => {
-            if (this.mode !== HubLED.Mode.COLOR) {
-                this.subscribe(HubLED.Mode.COLOR);
-            }
             if (typeof color === "boolean") {
                 color = 0;
             }
-            this.send(Buffer.from([0x81, this.portId, 0x11, 0x51, 0x00, color]));
+            this.sendWithMode("COLOR", Buffer.from([0x81, this.portId, 0x11, 0x51, 0x00, color]));
```

Comment:
The `0x81, this.portId, 0x11``is common to all devices sent messages. I am not sure to understand that part of the protocol yet but I think it refers to output commands.

`Device` could have some wrapper method to prepend this part and remove repetitions.

Reply:
Agreed and good spot - this is known in the LWP3 documentation as WriteDirect mode, so perhaps I'll create a method called `writeDirect`.